examples/island/Island.cpp

```diff
@@ -316,11 +326,11 @@ sf::Color getLowlandsTerrainColor(float moisture)
 {
     sf::Color color =
         moisture < 0.27f ? sf::Color(240, 240, 180) :
-        moisture < 0.3f ? sf::Color(240 - 240 * (moisture - 0.27f) / 0.03f, 240 - 40 * (moisture - 0.27f) / 0.03f, 180 - 180 * (moisture - 0.27f) / 0.03f) :
+        moisture < 0.3f ? sf::Color(240 - static_cast<sf::Uint8>(240 * (moisture - 0.27f) / 0.03f), 240 - static_cast<sf::Uint8>(40 * (moisture - 0.27f) / 0.03f), 180 - static_cast<sf::Uint8>(180 * (moisture - 0.27f) / 0.03f)) :
         moisture < 0.4f ? sf::Color(0, 200, 0) :
-        moisture < 0.48f ? sf::Color(0, 200 - 40 * (moisture - 0.4f) / 0.08f, 0) :
+        moisture < 0.48f ? sf::Color(0, 200 - static_cast<sf::Uint8>(40 * (moisture - 0.4f) / 0.08f), 0) :
         moisture < 0.6f ? sf::Color(0, 160, 0) :
-        moisture < 0.7f ? sf::Color(34 * (moisture - 0.6f) / 0.1f, 160 - 60 * (moisture - 0.6f) / 0.1f, 34 * (moisture - 0.6f) / 0.1f) :
+        moisture < 0.7f ? sf::Color(static_cast<sf::Uint8>(34 * (moisture - 0.6f) / 0.1f), 160 - static_cast<sf::Uint8>(60 * (moisture - 0.6f) / 0.1f), static_cast<sf::Uint8>(34 * (moisture - 0.6f) / 0.1f)) :
```

Comment:
While eliminating warnings, this makes code totally unreadable, and misses the point of the example.

It's a bit a weakness of C++ that a lot of valid code gets warnings. In theory, something like `short = short + short;` could already be problematic. And it's not like `static_cast` actually solves a real issue, it just silences the compiler.

Should we maybe add a `color()` function to the example that accepts `int`?
Or a `u8()` function as a shortcut for `static_cast<sf::Uint8>()`?
Or any other idea?

Reply:
You mean, a `color()` function that accepts `float`s, right? I guess, I can do that.
On the other hand it's an advanced example focusing on how to use Vulkan, so a few `static_cast` mixed in there, shouldn't really bother anyone...

Reply:
Yes, I meant `float`s.

True, but someone might be interested in the noise algorithm, and for that the `static_cast` is just distraction...